src/service/mercury/helpers/transformers.ts

```diff
@@ -863,23 +866,34 @@ const transformAccountHistory = async (
 ): Promise<Partial<Horizon.ServerApi.OperationRecord>[]> => {
   const invokeHostFnEdges =
     rawResponse.data?.invokeHostFnByPublicKey.edges || [];
-  const invokeHostFn = invokeHostFnEdges.map(
-    (edge) =>
-      ({
-        auth: edge.node.auth,
-        hostFunction: edge.node.hostFunction,
-        sorobanMeta: edge.node.sorobanMeta,
-        source: edge.node.source,
-        tx: edge.node.tx,
-        type: "invoke_host_function",
-        type_i: 24,
-        id: edge.node.opId,
-        transaction_attr: {
-          operation_count: edge.node.txInfoByTx.opCount,
-          fee_charged: edge.node.txInfoByTx.fee,
-        },
-      } as Partial<Horizon.ServerApi.InvokeHostFunctionOperationRecord>)
-  );
+  const invokeHostFn = invokeHostFnEdges.map((edge) => {
+    const hostFn = xdr.HostFunction.fromXDR(
+      Buffer.from(edge.node.hostFunction, "base64")
+    );
+    const invocation = hostFn.invokeContract();
```

Comment:
Not sure what your requirements are in terms of operations you display, but it's worth pointing out that host invocations can also be create contract and uploadcontractwasm.

Reply:
hey thanks for the info there, the client ignores those types for now but we may pull those into history in the future as well.